Grid.js

```diff
@@ -0,0 +1,139 @@
+import Cell from "./Cell.js";
+import { getRandomInt } from "./gridUtils.js";
+class Grid {
+    constructor(gridSize, numMines) {
+        this.gridSize = gridSize;
+        this.numMines = numMines;
+        this.cells = [];
+    }
+
+    createGridCells() {
+        this.cells = new Array();
+        for (let row = 0; row < this.gridSize; row++) {
+            for (let col = 0; col < this.gridSize; col++) {
+                let cell = new Cell(row, col);
+                this.cells.push(cell);
+            }        
+        }
+    }
+
+    getCellByRowCol (row, col){
+        let cell;
+        this.cells.forEach(c =>{
+            if (c.row == row && c.col == col){
+                cell = c;
+                //TODO: BREAK
+            }
+        })
+        return cell;
+    }
+    
+    assignMines(){ 
+        let numMinesCreated = 0;
+        while (numMinesCreated < this.numMines) {
+            let numGridCells = this.gridSize**2;
+            let randCell = getRandomInt(numGridCells - 1);
+            let cell = this.cells[randCell];
+            if (!cell.isMine) {
+                cell.isMine = true;
+                numMinesCreated++;
+            }
+        }
+    };
+
+    applyFnToNeighbors(cell, fn){
+        let row = cell.row;
+        let col = cell.col;
+
+        let topLeft = {row: row-1, col: col-1};
+        let topCenter = {row: row-1, col: col};
+        let topRight = {row: row-1, col: col+1};
+        
+        let centerLeft = {row: row, col: col-1};
+        let centerRight = {row: row, col: col+1};
+    
+        let bottomLeft = {row: row+1, col: col-1};
+        let bottomCenter = {row: row+1, col: col};
+        let bottomRight = {row: row+1, col: col+1};
+    
+        let neighborCoordinates = [topLeft, topCenter, topRight, 
+                                centerLeft, centerRight, 
+                                bottomLeft, bottomCenter, bottomRight];
+       
+        
+
+        const neighborCells = neighborCoordinates
+            .filter(c => this.isCellInBounds(c))
+            .map(c => this.getCellByRowCol(c.row, c.col));
+        neighborCells.forEach(neighborCell => fn(neighborCell));
+    };
+
+    isCoordinateInBounds(coordinate){
+        return coordinate >= 0 && coordinate < this.gridSize;
+    }
+
+    isCellInBounds(cell){
+        return this.isCoordinateInBounds(cell.row) && this.isCoordinateInBounds(cell.col);
+    }
+
+    isMineAndInBounds(surroundingCell) {
+        if (!this.isCellInBounds(surroundingCell)) {
+            return false;  
+        }
+        return surroundingCell.isMine;
+    }
+
+    updateThisNumMines(cell, neighborCell){
+        if(this.isMineAndInBounds(neighborCell))
+            cell.numMinesAround++;
+    }
+
+    updateEachCellNumMineAround(){
```

Comment:
make use of the `neighbor` convention that you have elsewhere, it's nice to be consistent w/naming throughout your code.
```suggestion
    updateAllMineNeighborCount(){
```

you might also want to change the prior function to match
```
    updateSingleMineNeighborCount()
```